extensions/sockets/src/main/java/org/mule/module/socket/api/client/SocketClient.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.module.socket.api.client;
+
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.api.connection.ConnectionValidationResult;
+
+public interface SocketClient
+{
+    boolean isValid();
+
+    ConnectionValidationResult validate();
```

Comment:
it doesn't make much sense to have a validate and a isValid method. It's also questionable if the validate() method should return a ConnectionValidationResult since that's a SDK specific class and could prevent this class to be externalized. 

I would just have a validate() method which is void and throws ConnectionException upon failure